workflow/controller/controller.go

```diff
@@ -505,11 +505,16 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 				return err
 			}
 		case labelPodCompleted:
+
 			_, err := pods.Patch(
 				ctx,
 				podName,
-				types.MergePatchType,
-				[]byte(`{"metadata": {"labels": {"workflows.argoproj.io/completed": "true"}}}`),
+				types.JSONPatchType,
+				[]byte(`[
+ {"op": "test", "path": "/metadata/labels/workflows.argoproj.io~completed", "value": "false"},
+ {"op": "remove", "path": "/metadata/finalizers/workflows.argoproj.io"}
```

Comment:
can we remove list elements by values in json patch, I don't think it's supported (not mentioned in the [RFC](https://www.rfc-editor.org/rfc/rfc6902))

Reply:
Yeah, its a shame. I think we can query the `wfc.podInformer` to make sure we don't remove the wrong finalizers.

Reply:
It's see it is `~1` not `~` to escape `/` in field names.